instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py

```diff
@@ -204,10 +211,48 @@ def collect_request_attributes(environ):
         flavor = flavor[len(_HTTP_VERSION_PREFIX) :]
     if flavor:
         result[SpanAttributes.HTTP_FLAVOR] = flavor
+    capture_custom_request_headers(
+        environ,
+        result,
+    )
 
     return result
 
 
+def capture_custom_request_headers(environ, attributes):
```

Comment:
Any specific reason to pass the attribute dict as a argument and modifying it with new headers rather than returning the headers to calling function?
I think it will be better from readability perspective to update the result dict in a single function rather than passing to called function and update there

Reply:
Agree. It is not clear what the function does without reading the function body.

Reply:
Added suggested changes to improve readability. Please have a look.